app/components/publications/publication/documents/documents-upload-modal.js

```diff
@@ -0,0 +1,86 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency-decorators';
+import { all } from 'ember-concurrency';
+import moment from 'moment';
+
+export default class PublicationsPublicationDocumentsDocumentsUploadModalComponent extends Component {
+  @inject store;
+
+  @tracked isExpanded = false;
+  @tracked newPieces = [];
+
+  @action
+  toggleSize() {
+    this.isExpanded = !this.isExpanded;
+  }
+
+  @action
+  uploadPiece(file) {
+    const now = moment().utc()
+      .toDate();
+    const documentContainer = this.store.createRecord('document-container', {
+      created: now,
+    });
+    const piece = this.store.createRecord('piece', {
+      created: now,
+      modified: now,
+      file: file,
+      accessLevel: this.defaultAccessLevel,
+      confidential: false,
+      name: file.filenameWithoutExtension,
+      documentContainer: documentContainer,
+    });
+    this.newPieces.pushObject(piece);
+  }
+
+  @task
+  *cancelUploadPieces() {
+    const deleteTasks = this.newPieces.map((piece) => this.deleteUploadedPiece.perform(piece));
```

Comment:
Zou `this.newPieces.map(this.deleteUploadedPiece.perform);` hier hetzelfde doen? :thinking:

Reply:
Ik heb het voorlopig zo gehouden, methods als functies gebruiken in JavaScript geeft vaak bugs.